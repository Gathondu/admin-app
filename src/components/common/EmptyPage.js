import React from "react";
import PropTypes from "prop-types";

const EmptyPage = ({text}) => {
  return (
    <div className="jumbotron">
      <i className="fa fa-ban fa-5x" aria-hidden="true" />
      <h2>{text}</h2>
    </div>
  );
};

  EmptyPage.propTypes = {
      text: PropTypes.string.isRequired
  };

  EmptyPage.defaultProps = {
      text: "No Courses to display. Add course to view"
  };

export default EmptyPage;
