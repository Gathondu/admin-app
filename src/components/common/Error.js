import React from "react";
import PropTypes from "prop-types";

const Error = ({ text, type }) => {
  return (
    <div className="jumbotron">
      <h1 key="type">{type}</h1>
      <h3 key="text">{text}</h3>
    </div>
  );
};

Error.propTypes = {
  type: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired
};

Error.defaultProps = {
  type: 404,
  text: "Page Not Found!"
};

export default Error;
