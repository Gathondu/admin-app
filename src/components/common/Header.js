import React from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";
import LoadingDots from "./LoadingDots";

const Header = ({ loading, totalCourses, totalAuthors }) => {
  return (
    <nav>
      <NavLink exact to="/" activeClassName="active">
        Home
      </NavLink>
      {" | "}
      <NavLink to="/authors" activeClassName="active">
        Authors
      </NavLink>
      <span>&nbsp;</span>
      <span>{totalAuthors}</span>
      {" | "}
      <NavLink to="/courses" activeClassName="active">
        Courses
      </NavLink>
      <span>&nbsp;</span>
      <span>{totalCourses}</span>
      {" | "}
      <NavLink to="/about" activeClassName="active">
        About
      </NavLink>
      {loading && <LoadingDots interval={100} dots={20} />}
    </nav>
  );
};

Header.propTypes = {
  loading: PropTypes.bool.isRequired,
  totalCourses: PropTypes.number.isRequired,
  totalAuthors: PropTypes.number.isRequired
};

export default Header;
