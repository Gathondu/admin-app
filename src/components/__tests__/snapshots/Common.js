import renderer from 'react-test-renderer';
import React from 'react';
import {shallow} from 'enzyme';
import {BrowserRouter} from 'react-router-dom';
import EmptyPage from '../../common/EmptyPage';
import Error from '../../common/Error';
import Header from '../../common/Header';
import SelectInput from '../../common/SelectInput';
import TextInput from '../../common/TextInput';
import LoadingDots from "../../common/LoadingDots";
import {authors} from "../../../api/mockAuthor";

describe('Common Components',()=>{
    describe('EmptyPage Component',()=>{
        it('should not regress',()=>{
            const tree = renderer.create(
                <EmptyPage text={'test text'}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('ErrorPage Component',()=>{
        it('should not regress',()=>{
            const tree = renderer.create(
                <Error type={1} text={'test error'}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('Header Component',()=>{
        it('should not regress',()=>{
            const tree = renderer.create(
                <BrowserRouter>
                    <Header loading={true} totalCourses={4} totalAuthors={4}/>
                </BrowserRouter>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('SelectInput Component',()=>{
        it('should not regress',()=>{
            const tree = renderer.create(
                <SelectInput options={[]} label={'test label'} name={'test button name'} onChange={()=>{}}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('should return options with given input', () => {
            const auths = authors.map(author => ({
                    value: author.id,
                    text: author.firstName + " " + author.lastName
                }));
            const props = {
                options: auths,
                label: 'test label',
                name: 'test button name',
                error: 'error',
                onChange: jest.fn()
            };
            const wrapper = shallow(<SelectInput {...props}/>);
            expect(wrapper.find('option')).toHaveLength(4);
        });
    });
    describe('TextInput Component',()=>{
        it('should not regress',()=>{
            const tree = renderer.create(
                <TextInput label={'test label'} onChange={()=>{}} name={'test input name'}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
        it('should show error', () => {
            const wrapper = shallow(<TextInput error={'error'} label={'test label'} onChange={()=>{}} name={'test input name'}/>);
            expect(wrapper.is('.has-error')).toBe(true);
        });
    });
    describe('LoadingDots Component', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <LoadingDots/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});