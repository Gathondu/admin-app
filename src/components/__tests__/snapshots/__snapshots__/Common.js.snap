// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Components EmptyPage Component should not regress 1`] = `
<div
  className="jumbotron"
>
  <i
    aria-hidden="true"
    className="fa fa-ban fa-5x"
  />
  <h2>
    test text
  </h2>
</div>
`;

exports[`Common Components ErrorPage Component should not regress 1`] = `
<div
  className="jumbotron"
>
  <h1>
    1
  </h1>
  <h3>
    test error
  </h3>
</div>
`;

exports[`Common Components Header Component should not regress 1`] = `
<nav>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
   | 
  <a
    aria-current={null}
    href="/authors"
    onClick={[Function]}
  >
    Authors
  </a>
  <span>
     
  </span>
  <span>
    4
  </span>
   | 
  <a
    aria-current={null}
    href="/courses"
    onClick={[Function]}
  >
    Courses
  </a>
  <span>
     
  </span>
  <span>
    4
  </span>
   | 
  <a
    aria-current={null}
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
</nav>
`;

exports[`Common Components LoadingDots Component should not regress 1`] = `
<span
  dots={3}
  interval={300}
>
  .
   
</span>
`;

exports[`Common Components SelectInput Component should not regress 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="test button name"
  >
    test label
  </label>
  <div
    className="field"
  >
    <select
      className="form-control"
      name="test button name"
      onChange={[Function]}
    >
      <option
        value=""
      />
    </select>
  </div>
</div>
`;

exports[`Common Components TextInput Component should not regress 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="test input name"
  >
    test label
  </label>
  <div
    className="field"
  >
    <input
      className="form-control"
      name="test input name"
      onChange={[Function]}
      type="text"
    />
  </div>
</div>
`;
